pkg/translate/translateValue.go

```diff
@@ -229,19 +247,135 @@ func (t *ValueYAMLTranslator) setEnablementAndNamespacesFromValue(valueSpec map[
+func translateHPASpec(inPath string, outPath string, value interface{}, valueTree map[string]interface{}, cpSpecTree map[string]interface{}) error {
+	asEnabled, ok := value.(bool)
+	if !ok {
+		return fmt.Errorf("expect autoscaleEnabled node type to be bool but got: %T", value)
+	}
+	if asEnabled {
+		newP := strings.Split(inPath, ".")[0]
+		minPath := newP + ".autoscaleMin"
+		asMin, found, err := name.GetFromTreePath(valueTree, util.ToYAMLPath(minPath))
+		if found && err == nil {
+			if err := tpath.WriteNode(cpSpecTree, util.ToYAMLPath(outPath+".minReplicas"), asMin); err != nil {
+				return err
+			}
+		}
+		maxPath := newP + ".autoscaleMax"
+		asMax, found, err := name.GetFromTreePath(valueTree, util.ToYAMLPath(maxPath))
+		if found && err == nil {
+			log.Infof("path has value in helm Value.yaml tree, mapping to output path %s", outPath)
+			if err := tpath.WriteNode(cpSpecTree, util.ToYAMLPath(outPath+".maxReplicas"), asMax); err != nil {
+				return err
+			}
+		}
+	}
+	return nil
+}
+
+// translateEnv translates env value from helm values.yaml tree
+func translateEnv(outPath string, value interface{}, cpSpecTree map[string]interface{}) error {
+	envMap, ok := value.(map[string]interface{})
+	if !ok {
+		return fmt.Errorf("expect env node type to be map[string]interface{} but got: %T", value)
+	}
+	outEnv := make([]map[string]interface{}, len(envMap))
+	cnt := 0
+	for k, v := range envMap {
+		outEnv[cnt] = make(map[string]interface{})
+		outEnv[cnt]["name"] = k
+		outEnv[cnt]["value"] = v
+		cnt++
+	}
+	log.Infof("path has value in helm Value.yaml tree, mapping to output path %s", outPath)
+	if err := tpath.WriteNode(cpSpecTree, util.ToYAMLPath(outPath), outEnv); err != nil {
+		return err
+	}
+	return nil
+}
+
+// translateAffinity translates Affinity related configurations from helm values.yaml tree
+func translateAffinity(outPath string, value interface{}, cpSpecTree map[string]interface{}) error {
+	affinityNode, ok := value.([]interface{})
+	if !ok {
+		return fmt.Errorf("expect affinity node type to be []interface{} but got: %T", affinityNode)
+	}
+	log.Infof("path has value in helm Value.yaml tree, mapping to output path %s", affinityNode)
+	if err := tpath.WriteNode(cpSpecTree, util.ToYAMLPath(outPath), affinityNode); err != nil {
+		return err
+	}
+	return nil
+}
+
+// translateK8sTree is internal method for translating K8s configurations from value.yaml tree.
+func translateK8sTree(valueTree map[string]interface{},
 	cpSpecTree map[string]interface{}, mapping map[string]*Translation) error {
 	for inPath, v := range mapping {
-		// Extra logics needed for K8s translation
-		// HPA spec
+		log.Infof("Checking for k8s path %s in helm Value.yaml tree", inPath)
+		m, found, err := name.GetFromTreePath(valueTree, util.ToYAMLPath(inPath))
+		if err != nil {
+			return err
+		}
+		if !found {
+			log.Infof("path %s not found in helm Value.yaml tree, skip mapping.", inPath)
+			continue
+		}
+		iP := strings.Split(inPath, ".")
```

Comment:
should be ip - in Go, for acronyms keep the case consistent depending on the first letter e.g. 
HTTPVar, httpVar are ok
hTTPVar is not ok